### PR TITLE
Add optional python dependencies to snap packaging

### PR DESCRIPTION
**Pull request by devec0**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#changelog)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

This PR adds the optional dependencies to snapcraft packaging, including required build and stage libraries to enable features such as PostgreSQL support.

NOTE: This is currently done by adding the '.[all]' requirement to the python-packages section of the snapcraft file, to avoid having to keep the snapcraft file up to date as dependencies change, and also to work around al imitation in snapcraft, noted in [LP#1786939](https://bugs.launchpad.net/snapcraft/+bug/1786939). Once this bug is fixed, then the method could be changed to use a supported snapcraft option to pull in additional dependencies based on the `extras_require` section of setup.py directly.

The additional stage and build packages are just used to ensure the required binary libraries which are needed by the optional dependencies are included in the resulting snap.

I have built and tested t his snap and am able to run a synapse instance using PostgreSQL as backing database based on these changes.

Signed-off-by: James Hebden <james@ec0.io>

### CODE REVIEW

File: changelog.d/6317.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Add optional python dependencies and dependant binary libraries to snapcraft packaging
```
Comment (anoadragon453) [inline changelog.d/6317.misc]:
```suggestion
Add optional python dependencies and dependant binary libraries to snapcraft packaging.
```

Changelog entries need to end in a `.` or `!`.

**Comment by devec0** [comment]
Thanks for the review! I've updated the changelog entry to meet requirements.

File: changelog.d/6317.misc (hunk 1 above)
Comment (devec0) [reply changelog.d/6317.misc]: Done, and thanks

**Comment by anoadragon453** [comment]
Looks good, thank you!

**Pull request merged**